packages/blade/src/components/Dropdown/DropdownOverlay.web.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import styled, { keyframes, css } from 'styled-components';
+import type { FlattenSimpleInterpolation } from 'styled-components';
+import { useDropdown } from './useDropdown';
+import Box from '~components/Box';
+import { makeMotionTime } from '~utils';
+import { useTheme } from '~components/BladeProvider';
+
+const dropdownFadeIn = keyframes`
+from {
+  transform: translateY(0px);
+  opacity: 0;
+}
+
+to {
+  transform: translateY(8px);
+  opacity: 1;
+}
+`;
+
+const dropdownFadeOut = keyframes`
+from {
+  transform: translateY(8px);
+  opacity: 1;
+}
+
+to {
+  transform: translateY(0px);
+  opacity: 0;
+}
+`;
+
+const StyledDropdownOverlay = styled(Box)<{
+  transition: FlattenSimpleInterpolation;
+}>(
+  (props) =>
+    css`
+      ${props.transition}
+      transform: translateY(8px);
+      opacity: 0;
+    `,
+);
+
+/**
+ * Overlay for dropdown.
+ *
+ * Wrap your ActionList with this this component
+ */
+function DropdownOverlay({ children }: { children: React.ReactNode }): JSX.Element {
+  const { isOpen } = useDropdown();
+  const { theme } = useTheme();
+  const [display, setDisplay] = React.useState<'none' | 'block'>('none');
+
+  const fadeIn = css`
+    animation: ${dropdownFadeIn} ${makeMotionTime(theme.motion.duration.quick)}
+      ${theme.motion.easing.entrance.revealing as string};
+  `;
+
+  const fadeOut = css`
+    animation: ${dropdownFadeOut} ${makeMotionTime(theme.motion.duration.quick)}
+      ${theme.motion.easing.entrance.revealing as string};
+  `;
+
+  React.useEffect(() => {
+    if (isOpen) {
+      setDisplay('block');
+    }
+  }, [isOpen]);
+
+  return (
+    <Box position="relative">
+      <StyledDropdownOverlay
+        style={{ opacity: isOpen ? 1 : 0 }}
+        display={display}
+        position="absolute"
+        transition={isOpen ? fadeIn : fadeOut}
+        onAnimationEnd={() => {
+          if (isOpen) {
+            setDisplay('block');
+          }
+
+          if (!isOpen) {
+            setDisplay('none');
+          }
```

Comment:
```ts
          if (isOpen) {
            setDisplay('block');
          } else {
            setDisplay('none');
          }
          
```

or

```
setDisplay(isOpen ? 'block' : 'none');
```

😅

Reply:
lol yes will do. Was probably 🍻 while writing this

Reply:
done